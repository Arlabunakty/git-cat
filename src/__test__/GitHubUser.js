export const user = {
  login: "Arlabunakty",
  id: 3165275,
  node_id: "MDQ6VXNlcjMxNjUyNzU=",
  avatar_url: "https://avatars.githubusercontent.com/u/3165275?v=4",
  gravatar_id: "",
  url: "https://api.github.com/users/Arlabunakty",
  html_url: "https://github.com/Arlabunakty",
  followers_url: "https://api.github.com/users/Arlabunakty/followers",
  following_url:
    "https://api.github.com/users/Arlabunakty/following{/other_user}",
  gists_url: "https://api.github.com/users/Arlabunakty/gists{/gist_id}",
  starred_url:
    "https://api.github.com/users/Arlabunakty/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/Arlabunakty/subscriptions",
  organizations_url: "https://api.github.com/users/Arlabunakty/orgs",
  repos_url: "https://api.github.com/users/Arlabunakty/repos",
  events_url: "https://api.github.com/users/Arlabunakty/events{/privacy}",
  received_events_url:
    "https://api.github.com/users/Arlabunakty/received_events",
  type: "User",
  site_admin: false,
  name: "Denys Durniev",
  company: null,
  blog: "",
  location: "Ukraine",
  email: null,
  hireable: null,
  bio: null,
  twitter_username: null,
  public_repos: 12,
  public_gists: 4,
  followers: 2,
  following: 2,
  created_at: "2013-01-01T21:35:36Z",
  updated_at: "2022-06-09T17:39:59Z",
};
