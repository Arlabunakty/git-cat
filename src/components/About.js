import React from "react";

const About = () => {
  return <h1 data-testid="about">About Component</h1>;
};

export default About;
