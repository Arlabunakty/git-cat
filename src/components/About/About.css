.about-container {
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
  padding: 2em 2em;
  margin: 0;
  list-style: none;
  column-gap: 3em;
}

.about-avatar {
  height: 200px;
  border-radius: 50%;
  overflow: hidden;
}

.about-text {
  display: flex;
  flex-flow: column wrap;
}

.account-details {
  color: #4c56c0;
}
