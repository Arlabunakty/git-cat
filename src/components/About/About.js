import React from "react";
import "./About.css";

const fields = [
  { name: "bio", propertyName: "bio" },
  { name: "country", propertyName: "location" },
  { name: "company", propertyName: "company" },
  { name: "email", propertyName: "email" },
  { name: "profile", propertyName: "html_url", link: true },
  { name: "subscriptions", propertyName: "subscriptions_url", link: true },
  { name: "received events", propertyName: "received_events_url", link: true },
  { name: "repos", propertyName: "repos_url", link: true },
  { name: "followers", propertyName: "followers_url", link: true },
];

class About extends React.Component {
  render() {
    return (
      <div className="about-container">
        <img
          className="about-avatar"
          src={this.props.user.avatar_url}
          alt="avatar"
        />
        <div className="about-text">
          <h1>{this.props.user.name}</h1>
          <h3 className="account-details">ACCOUNT DETAILS</h3>
          {fields.map((f) => (
            <span key={f.name}>
              <b>{f.name}: </b>
              {f.link ? (
                <a href={this.props.user[f.propertyName]}>link</a>
              ) : (
                this.props.user[f.propertyName]
              )}
            </span>
          ))}
        </div>
      </div>
    );
  }
}

export default About;
