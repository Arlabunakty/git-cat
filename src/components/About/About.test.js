import React from "react";
import renderer from "react-test-renderer";
import About from "./About";
import { user } from "./../../__test__/GitHubUser";

test("display user statistics", async () => {
  const tree = renderer.create(<About user={user} />).toJSON();

  expect(tree).toMatchSnapshot();
});
