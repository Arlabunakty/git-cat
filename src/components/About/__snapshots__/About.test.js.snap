// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display user statistics 1`] = `
<div
  className="about-container"
>
  <img
    alt="avatar"
    className="about-avatar"
    src="https://avatars.githubusercontent.com/u/3165275?v=4"
  />
  <div
    className="about-text"
  >
    <h1>
      Denys Durniev
    </h1>
    <h3
      className="account-details"
    >
      ACCOUNT DETAILS
    </h3>
    <span>
      <b>
        bio
        : 
      </b>
    </span>
    <span>
      <b>
        country
        : 
      </b>
      Ukraine
    </span>
    <span>
      <b>
        company
        : 
      </b>
    </span>
    <span>
      <b>
        email
        : 
      </b>
    </span>
    <span>
      <b>
        profile
        : 
      </b>
      <a
        href="https://github.com/Arlabunakty"
      >
        link
      </a>
    </span>
    <span>
      <b>
        subscriptions
        : 
      </b>
      <a
        href="https://api.github.com/users/Arlabunakty/subscriptions"
      >
        link
      </a>
    </span>
    <span>
      <b>
        received events
        : 
      </b>
      <a
        href="https://api.github.com/users/Arlabunakty/received_events"
      >
        link
      </a>
    </span>
    <span>
      <b>
        repos
        : 
      </b>
      <a
        href="https://api.github.com/users/Arlabunakty/repos"
      >
        link
      </a>
    </span>
    <span>
      <b>
        followers
        : 
      </b>
      <a
        href="https://api.github.com/users/Arlabunakty/followers"
      >
        link
      </a>
    </span>
  </div>
</div>
`;
